torchao/prototype/awq/core.py

```diff
@@ -0,0 +1,272 @@
+from dataclasses import dataclass
+from typing import Tuple, Optional
+
+import torch
+import torch.nn.functional as F
+from torch.utils._python_dispatch import return_and_correct_aliasing
+from torchao.dtypes.uintx.uintx import to_uintx
+from torchao.dtypes.affine_quantized_tensor import (
+    to_affine_quantized_intx,
+    LayoutType,
+    register_layout_cls,
+    AQTLayout,
+    register_aqt_quantized_linear_dispatch
+
+) 
+from torchao.quantization.quant_primitives import (
+    MappingType,
+    ZeroPointDomain,
+)
+from torchao.quantization.observer import (
+    AffineQuantizedObserverBase, PerGroup
+)
+
+
+class AWQObserver(AffineQuantizedObserverBase):
+    def __init__(self,
+        weight: torch.Tensor,
+        bias: torch.Tensor,
+        block_size: Tuple,
```

Comment:
nit: can this be https://github.com/pytorch/ao/blob/20f379608a8827087315789e636ad91d1e01d7ae/torchao/quantization/observer.py#L142

Reply:
Is this going to be used moving forward in other quantization methods as well